lib/cobra_commander/component_tree.rb

```diff
@@ -163,7 +163,21 @@ def package_json_path
       end
 
       def combined_deps(json)
-        Hash(json["dependencies"]).merge(Hash(json["devDependencies"]))
+        worskpace_dependencies = build_workspaces(json["workspaces"])
+        dependencies = format_dependencies Hash(json["dependencies"]).merge(Hash(json["devDependencies"]))
+        dependencies.push(worskpace_dependencies).flatten!.uniq!
+        dependencies
+      end
+
+      def build_workspaces(workspaces)
+        return [] if workspaces.nil?
+        workspaces.map do |workspace|
+          glob = "#{workspace}/package.json"
+          worskpace_dependencies = Dir.glob(glob)
```

Comment:
Will this ever have more than one entry? Can we avoid the `flatten` by only considering the first result?

Reply:
the way I understand, yes it can have more than one entry. The `workspaces` section contains an array of node components, with each item being the direct path to a component or a glob pattern that can represent more that on component. That's why we have to flatten it at the end so we can get a single list with all the components.